publish/kafka/publish.go

```diff
@@ -279,12 +279,12 @@ func (m *mtPublisher) Publish(metrics []*schema.MetricData) error {
 			if m.autoInterval {
 				_, s := m.schemas.Match(metric.Name, 0)
 				metric.Interval = s.Retentions.Rets[0].SecondsPerPoint
-				metric.SetId()
 			} else {
 				log.Error("interval is 0 but can't deduce interval automatically. this should never happen")
 				return errors.New("need to deduce interval but cannot")
 			}
 		}
+		metric.SetId()
```

Comment:
this looks like a bugfix sneaking in under a commit named "Add mt-parrot command to generate deterministic..."

what bug does this fix exactly? this warrants a bit more info, e.g. in a dedicated commit or even PR

Reply:
yeah, I'll toss another PR in for that...
Ingestion was failing because the metric ID was never set going through that code path

Reply:
https://github.com/grafana/metrictank/pull/1687